### PR TITLE
fix: mega menu differences in design

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/jira/software/c/projects/SFUI2/boards/69?modal=detail&selectedIssue=SFUI2-1043&assignee=5fcf86fca1d3f100684db8a5

# Scope of work

Adjusted mega menu to figma design

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx [hunk 1]
```diff
@@ -141,17 +153,21 @@ export default function BaseMegaMenu() {
...
-              <img src={brandLogoSign.src} alt="Sf Logo" className="w-8 h-8 mr-4 md:w-[12.5rem] md:h-[1.75rem]" />
+              <source srcSet={brandLogo.src} media="(min-width: 1025px)" />
+              <img src={brandLogoSign.src} alt="Sf Logo" className="w-8 h-8 lg:w-[12.5rem] lg:h-[1.75rem]" />
             </picture>
           </a>
-          <nav className="flex w-full justify-between flex-nowrap" aria-label="SF Navigation">
+          <nav className="flex justify-between w-full flex-nowrap" aria-label="SF Navigation">
             <ul>
               <li role="none">
                 <SfButton
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:169]: I think this button should get `square` prop on mobile. Right now its padding is a bit too big for an icon.
Screenshot 2023-04-21 at 12 10 59

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx [hunk 2]
```diff
@@ -219,9 +235,9 @@ export default function BaseMegaMenu() {
                         </ul>
                       </div>
                     ))}
-                    <div className="flex flex-col items-center justify-center bg-neutral-100 rounded-md border-neutral-300 overflow-hidden grow">
+                    <div className="flex flex-col items-center justify-center overflow-hidden rounded-md bg-neutral-100 border-neutral-300 grow">
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:238]: Banner is rounded on mobile as well, but it not supposed to be.
Screenshot 2023-04-21 at 12 09 47

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx [hunk 3]
```diff
@@ -219,9 +235,9 @@ export default function BaseMegaMenu() {
...
                       <img src={bannerDetails.image.src} alt={bannerDetails.title} className="object-contain" />
-                      <p className="mb-4 mt-4 px-4 text-center typography-text-base font-medium">
+                      <p className="px-4 mt-4 mb-4 font-medium text-center typography-text-base">
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:240]: Do you have an util like Prettier for Tailwind classes sorting enabled? Does it sort classes in a particular order? We may consider using it as the team to unify and prevent this kinda "changes" in the future.

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx [hunk 4]
```diff
@@ -141,17 +153,21 @@ export default function BaseMegaMenu() {
...
-          <a href="/" aria-label="SF Homepage" className="inline-block text-white mr-2 lg:mr-10">
+          <a
+            href="/"
+            aria-label="SF Homepage"
+            className="inline-block mr-2 text-white md:mr-10 focus-visible:outline focus-visible:outline-offset focus-visible:rounded-sm"
+          >
             <picture>
-              <source srcSet={brandLogo.src} media="(min-width: 767px)" />
-              <img src={brandLogoSign.src} alt="Sf Logo" className="w-8 h-8 mr-4 md:w-[12.5rem] md:h-[1.75rem]" />
+              <source srcSet={brandLogo.src} media="(min-width: 1025px)" />
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:162]:
```suggestion
              <source srcSet={brandLogo.src} media="(min-width: 1024px)" />
```
In order to have seamless transition on resizing we substract px on max-width so `max-width: 1023px` and `min-width: 1024px`

image

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx (hunk 3 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:240]: Yes, I used to have it. It was removed form this workspace but it seems it's enabled again. I'll check it and remove

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx (hunk 1 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:169]: It's exactly like on figma:  
Screenshot 2023-04-24 at 09 30 26

Comment (justyna-13) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:168]: https://github.com/vuestorefront/storefront-ui/pull/2622/commits/2a200c8395073393be3f5853f9162e9c2d8ea3d3

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx (hunk 4 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2622/commits/2a200c8395073393be3f5853f9162e9c2d8ea3d3

File: apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx (hunk 3 above)
Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:234]: I have it as well but I guess its better to not have things like that

Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:234]: @AdamPawlinski /\ :)

Comment (dkacper) [reply apps/preview/next/pages/showcases/MegaMenu/BaseMegaMenu.tsx:236]: I mean, imo it'd be nice to have it in the project, so it'd force all the classes to look exactly the same no matter of the author. 🤔

**Review by dkacper — APPROVED** [review]

**Pull request merged**